util/settings/settings.go

```diff
@@ -394,6 +394,15 @@ const (
 	settingsPasswordPatternKey = "passwordPattern"
 )
 
+var (
+	sourceTypeToEnableGenerationKey = map[v1alpha1.ApplicationSourceType]string{
+		v1alpha1.ApplicationSourceTypeKustomize: "kustomize.enable",
+		v1alpha1.ApplicationSourceTypeHelm:      "helm.enable",
+		v1alpha1.ApplicationSourceTypeDirectory: "jsonnet.enable",
+		v1alpha1.ApplicationSourceTypeKsonnet:   "ksonnet.enable",
```

Comment:
Ksonnet option can be removed as we are removing as part of https://github.com/argoproj/argo-cd/pull/8479? 

cc @jessesuen @gdsoumya

Reply:
Correct, but this change is intended for v2.3 release to provide users the ability to disable tools and improve security.